src/main/java/io/dropwizard/kafka/KafkaClientFactory.java

```diff
@@ -30,6 +30,12 @@
     @JsonProperty
     protected Set<String> bootstrapServers;
 
+    @JsonProperty
+    protected Optional<String> clientDNSLookup = Optional.empty();
```

Comment:
I'd prefer these just be in the following format, if you don't mind:

```
@JsonProperty
protected String clientDNSLookup;

@JsonProperty
protected String clientId;
````

And then perform if checks before setting the values.

Reply:
I'm basing these Optionals off of other fields in the class `KafkaProducerFactory` such as `acks`, `retries`, and `maxInFlightRequestsPerConnection` which are evaluated in `createBaseKafkaConfigurations()` like so: 
```
retries.ifPresent(retriesValue -> config.put(ProducerConfig.RETRIES_CONFIG, retriesValue));
```

The if check does exist for `metricsEnabled` but it seemed more consistent with the rest of the code to check the optional values in the style above.

Wanted to double check with you if that's preferable or if I should go ahead and make the suggested change. 

Thanks again for taking a look!

Reply:
Ah, I forgot I had used Optionals there back when I wrote the first (internal) version of this. It's not a big deal, so keeping it as you have it is fine for these.